src/renderer/components/SwarmDeployment.tsx

```diff
@@ -0,0 +1,229 @@
+  const popupStartDiv = (
+    <div id="initialize-swarm">
+      <label htmlFor="stack-name" id="stack-name-label">Stack Name</label>
+      <input id="stack-name" name="stack-name" placeholder="Enter name...." onChange={(event) => { stackNameRef.current = event.target.value }}></input>
+      <button 
+        id="create-swarm" 
+        onClick={() => { 
+          if (currentFile) {
+            console.log('stackName inside onClick: ', stackNameRef.current);
+            getNameAndDeploy(event)
+          } else {
+            setSuccess(false);
+            setNoFile(true);
+            setSwarmDeployState(0);
+          }
+        }}>
+        Create Swarm
+      </button>
+    </div>);
+  
+  // render this div if successful joining swarm
+  const successDiv = (
+    <div className="success-div">
+      <p className="success-p">
+        <span className="swarm-spans">Success! Your swarm has been deployed!</span>
+        <br></br>The current node {nodeAddress}<br></br>is now a manager
+      </p>
+    </div>);
+
+  // if unsuccessful / if no active file, render error dive                          
+  const errorDiv = (
+    <div className="error-div">
+      <p className="error-p">
+        Sorry, there was an issue initializing your swarm
+      </p>
+      <button
+        className="swarm-btn" 
+        onClick={() => {
+          leaveSwarm();
+        }}>Try Again
+      </button>
+    </div>);
+
+  // change visibility of HTML element from hidden to visible or vice versa
+  // used for the popup box
+  const toggleVisible = (element: any) => {
+    if (element) element.style.visibility = 'visible';
+  }
+  const toggleHidden = (element: any) => {
+    if (element) element.style.visibility = 'hidden';
+  }
+
+  // retrieve input from user and pass it to runDockerSwarmDeployment as an argument
+  // the function will return stdout from running each function, so that we have access to that information
+  const getNameAndDeploy = async (event: any) => {
+    // // get value from user's input
+    // console.log('current stack name from state: ', stackNameRef.current);
+
+    // hide pop-up while running commands
+    toggleHidden(swarmDeployPopup);
+    setSwarmDeployState(2);
+
+    // await results from running dwarm deployment shell tasks 
+    const returnedFromPromise = await runDockerSwarmDeployment(currentFile, stackNameRef.current);
+    const infoReturned = JSON.parse(returnedFromPromise);
+    console.log(infoReturned);
+    setInfoFromSwarm(infoReturned);
+
+    // if there is no error on the returned object, swarm initialisation was successful 
+    if (!infoReturned.init.error) {
+      setStdOutMessage(infoReturned.init.out.split('\n')[0]);
+      console.log(stdOutMessage);
+      console.log(infoFromSwarm);
+      // the split here is to get just the 25-character node ID of the swarm
+      setNodeAddress(infoReturned.init.out.split('\n')[0].split(' ')[4].replace(/[()]/g, ''));
```

Comment:
why not assign infoReturned.init.out.split('\n')[0] to a variable and then regex it to make this easier to read